File: Ahorn/entities/FinalTitanSummitBackgroundManager.jl
module FinalTitanSummitBackgroundManager

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/FinalTitanSummitBackgroundManager" FinalTitanSummitBackgroundManager(x::Integer, y::Integer, ambience::String="", cloudStrengthMultiplier::Number=1.0, creatureStrengthMultiplier::Number=1.0, cutscene::String="", dark::Bool=false, debrisStrengthMultiplier::Number=1.0, giygasStrengthMultiplier::Number=1.0, index::Integer=0, intro_launch::Bool=false, thunderStrengthMultiplier::Number=1.0)

const placements = Ahorn.PlacementDict(
    "FinalTitanSummitBackgroundManager" => Ahorn.EntityPlacement(FinalTitanSummitBackgroundManager)
)

function Ahorn.selection(entity::FinalTitanSummitBackgroundManager)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::FinalTitanSummitBackgroundManager, room::Maple.Room)
    Ahorn.drawSprite(ctx, "@Internal@/summit_background_manager", entity.x, entity.y)
end

end

File: Loenn/entities/FinalTitanSummitBackgroundManager.lua
local finalTitanSummitBackgroundManager = {}

finalTitanSummitBackgroundManager.name = "MaggyHelper/FinalTitanSummitBackgroundManager"
finalTitanSummitBackgroundManager.depth = 8900
finalTitanSummitBackgroundManager.texture = "@Internal@/summit_background_manager"
finalTitanSummitBackgroundManager.fieldInformation = {
    index = {
        fieldType = "integer",
        minimumValue = 0,
        maximumValue = 12
    },
    cloudStrengthMultiplier = {
        fieldType = "number",
        minimumValue = 0.0,
        maximumValue = 10.0
    },
    debrisStrengthMultiplier = {
        fieldType = "number",
        minimumValue = 0.0,
        maximumValue = 10.0
    },
    creatureStrengthMultiplier = {
        fieldType = "number",
        minimumValue = 0.0,
        maximumValue = 10.0
    },
    giygasStrengthMultiplier = {
        fieldType = "number",
        minimumValue = 0.0,
        maximumValue = 10.0
    },
    thunderStrengthMultiplier = {
        fieldType = "number",
        minimumValue = 0.0,
        maximumValue = 10.0
    },
    cutscene = {
        fieldType = "string"
    },
    intro_launch = {
        fieldType = "boolean"
    },
    dark = {
        fieldType = "boolean"
    },
    ambience = {
        fieldType = "string"
    }
}
finalTitanSummitBackgroundManager.fieldOrder = {
    "x", "y",
    "width", "height",
    "index",
    "cloudStrengthMultiplier",
    "debrisStrengthMultiplier",
    "creatureStrengthMultiplier",
    "giygasStrengthMultiplier",
    "thunderStrengthMultiplier",
    "cutscene",
    "intro_launch",
    "dark",
    "ambience"
}
finalTitanSummitBackgroundManager.placements = {
    {
        name = "FinalTitanSummitBackgroundManager",
        data = {
            width = 320,
            height = 180,
            index = 0,
            cloudStrengthMultiplier = 1.0,
            debrisStrengthMultiplier = 1.0,
            creatureStrengthMultiplier = 1.0,
            giygasStrengthMultiplier = 1.0,
            thunderStrengthMultiplier = 1.0,
            cutscene = "",
            intro_launch = false,
            dark = false,
            ambience = ""
        }
    }
}

return finalTitanSummitBackgroundManager
